Modules/_cursesmodule.c

```diff
@@ -233,11 +233,18 @@ static int
 PyCurses_ConvertToChtype(PyCursesWindowObject *win, PyObject *obj, chtype *ch)
 {
     long value;
-    if(PyBytes_Check(obj) && PyBytes_Size(obj) == 1) {
+    if (PyBytes_Check(obj)) {
+        if (PyBytes_GET_SIZE(obj) != 1) {
+            PyErr_Format(PyExc_TypeError,
+                         "expect bytes or str of length 1, or int, "
+                         "got a bytes of length %zd",
+                         PyBytes_GET_SIZE(obj));
+            return 0;
+        }
         value = (unsigned char)PyBytes_AsString(obj)[0];
     }
     else if (PyUnicode_Check(obj)) {
-        if (PyUnicode_GetLength(obj) != 1) {
+        if (PyUnicode_GET_LENGTH(obj) != 1) {
             PyErr_Format(PyExc_TypeError,
                          "expect bytes or str of length 1, or int, "
```

Comment:
In an "A or B" construction, it is an error to add a comma before 'or'.  But it is awkward when 'A' itself is an 'or' construction.  Given that 'B' is a single item, I think the best solution is to reverse the order.
```suggestion
                         "expect int or bytes or str of length 1,"
```
I think this is clear enough without an extra comma.  Same comment for the bytes addition below.

Reply:
Resolved by reordering iterms.